pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -636,7 +636,7 @@ private void setTimestampIndexExpressionOverrideHints(@Nullable Expression expre
             && timeExpression.getIdentifier() != null) {
           String timeColumn = timeExpression.getIdentifier().getName();
           String timeColumnWithGranularity = TimestampIndexGranularity.getColumnNameWithGranularity(timeColumn,
-              TimestampIndexGranularity.valueOf(granularString));
+              TimestampIndexGranularity.valueOf(granularString.toUpperCase()));
```

Comment:
Good catch. Suggest changing line 631 to directly store upper case as `granularString` which can avoid one extra conversion in `TimestampIndexGranularity.isValidTimeGranularity(granularString)`

Reply:
Yeh that makes sense. Have added a commit to do that.